tests/core/test_jose/test_jwe.py

```diff
@@ -886,6 +886,73 @@ def test_ecdh_1pu_jwe_with_okp_keys_in_key_agreement_with_key_wrapping_mode(self
                 rv = jwe.deserialize_compact(data, bob_key, sender_key=alice_key)
                 self.assertEqual(rv['payload'], b'hello')
 
+    def test_ecdh_1pu_jwe_json_serialization(self):
```

Comment:
Please add more tests for JSON serialization and multi-recipient.
Please add tests for the test vectors from ECDH-1PU and ECDH-ES where JSON serialization is used.

Reply:
Also, please add tests with edge cases, where unexpected parameters are passed:
- keys from different curves
- invalid/unexpected/missing values in the input header
- etc.

Reply:
Done.